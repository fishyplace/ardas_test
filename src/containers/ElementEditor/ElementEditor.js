import React from 'react';
import './ElementEditor.css';
import EditableField from './EditableField/EditableField';
import { Link } from 'react-router-dom';

const elementEditor = props => {

    const editTriggeredHandler = (id, name) => {
        props.editEntryHandler(id, name);
    }

    const {id} = props.match.params;
    const toEdit = props.data.find(el => el.id === +id);

    return (
        <div className="editor-container">
            <EditableField editEntryHandler={name => editTriggeredHandler(id, name)} name={toEdit.name}>Name : {toEdit.name}</EditableField>
            <p>Description: <br /> {toEdit.description}</p>
            <p>Estimated: {toEdit.estimated_effort}</p>
            <p>Actual: {toEdit.actual_effort}</p>
            <p>Due Date: {toEdit.due_date}</p>
            <p>Tags: {toEdit.tags ? toEdit.tags.join(' , ') : 'none'}</p>
            <Link to="/">Back</Link>
        </div>
    )
}

export default elementEditor;