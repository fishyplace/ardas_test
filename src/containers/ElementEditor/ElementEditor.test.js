import React from 'react';
import ElementEditor from './ElementEditor';

import { shallow } from 'enzyme';

it('should render without crashing', () => {

    const id = '3';
    const data = [{
        id: 3,
        name: "Today_task1",
        creation_date: "2015-04-21T06:50:21",
        due_date: "2015-04-22T23:59:00",
        start_date: "2015-04-21T00:00:01",
        is_completed: false,
        is_archived: false,
        estimated_effort: 5.5,
        actual_effort: 3.3,
        physical_progress: 60,
        obj_status: "active",
        description: "Lorem_ipsum_dolor_sit_amet,_consectetur_adipiscing_elit",
        project_id: 0
      }];
    const match = {
        params: {
            id
        }
    }
    
    const component = shallow(<ElementEditor match={match} data={data} />);
    expect(component).toMatchSnapshot();
})