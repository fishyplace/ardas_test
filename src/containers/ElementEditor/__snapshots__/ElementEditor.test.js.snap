// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="editor-container"
>
  <EditableField
    editEntryHandler={[Function]}
    name="Today_task1"
  >
    Name : 
    Today_task1
  </EditableField>
  <p>
    Description: 
    <br />
     
    Lorem_ipsum_dolor_sit_amet,_consectetur_adipiscing_elit
  </p>
  <p>
    Estimated: 
    5.5
  </p>
  <p>
    Actual: 
    3.3
  </p>
  <p>
    Due Date: 
    2015-04-22T23:59:00
  </p>
  <p>
    Tags: 
    none
  </p>
  <Link
    replace={false}
    to="/"
  >
    Back
  </Link>
</div>
`;
